pkg/provider/aws/plugin.go

```diff
@@ -23,7 +23,7 @@ func (a *AWS) ExpandConstructs(result *core.ConstructGraph, dag *core.ResourceGr
 			merr.Append(a.expandExpose(dag, construct))
 		case *core.Orm:
 			merr.Append(a.expandOrm(dag, construct))
-		case *core.Fs:
+		case *core.Fs, *core.InternalResource:
```

Comment:
Are all `InternalResource`s Fs? Seems like this is too generically named or that there's a field or something on there that should indicate how to expand it.

Reply:
right now they are since its just the payload bucket. Open to other suggestions just didnt want to make things too complex

Reply:
Also, that seems like an AWS-specific detail (the payloads bucket). Is there a reason we don't just create the S3 bucket as part of this plugin? Doesn't make sense as a construct IMO. That said, may be beyond the scope of this change if it's too much, so feel free to disregard (maybe ticket for it though).

Reply:
im a little confused on the last message. Agreed we need to work on the internal payloads bucket, but what exactly do we want the ticket to be about?